src/System.Text.Json/tests/Serialization/Object.ReadTests.cs

```diff
@@ -356,5 +356,22 @@ public static void ReadStructWithSimpleClassArrayValueTest()
             Assert.Equal(1, parsedObject.MySimpleTestClass.MyInt32Array[0]);
             Assert.Equal(2, parsedObject.MyInt32Array[0]);
         }
+
+        public class ClassWithNoSetter
```

Comment:
Can we add some more test permutations. For instance, add a test class with more settable/non-settable properties (rather than just 1 of each). Since the property ordering caused the original issue, it would be good to have something like this:

```C#
public class ClassWithNonSettableProperties
{
   public SimpleTestClass SkippedChild { get; }
   public SimpleTestClass ParsedChild { get; set; }
   public SimpleTestClass AnotherSkippedChild { get; }
   public SimpleTestClass AnotherParsedChild { get; set; }
}
```

Reply:
@ahsonkhan I improved the individual tests for arrays, dictionaries, & objects. Especially the json fed to the (de)serializer, which now includes stuff to be drained, invalid elements, duplicates, etc.